mypy/parse.py

```diff
@@ -1853,7 +1856,7 @@ def current_str(self) -> str:
     def peek(self) -> Token:
         return self.tok[self.ind + 1]
 
-    def parse_error(self) -> None:
+    def parse_error(self) -> ThrowsException:
```

Comment:
This is super confusing. The function raises, but is also declared to return an exception, and its callers raise whatever it returns, but those raise statements are never raised because the called function raises. Wouldn't it be simpler to make this function *actually* return the exception and have the call sites *actually* raise the exception it returns?